src/mito2/src/read/compat.rs

```diff
@@ -85,7 +86,7 @@ pub(crate) fn has_same_columns(left: &RegionMetadata, right: &RegionMetadata) ->
     }
 
     for (left_col, right_col) in left.column_metadatas.iter().zip(&right.column_metadatas) {
-        if left_col.column_id != right_col.column_id {
+        if left_col != right_col {
```

Comment:
Only comparing what you need can be a little faster. And the `ColumnMetadata` contains much information we don't care about here like name, metadata, semantic type etc. Can you make a util method to only compare whether we need cast types on read?